vip-helpers/vip-newrelic.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/**
+ * Disable New Relic's browser metrics
+ *
+ * Removes NR's JavaScript for tracking browser metrics, including page load times, Apdex score, and more.
+ *
+ * Must be called at or before the `template_redirect` action.
+ */
+function wpcom_vip_disable_new_relic_js() {
+	if ( did_action( 'template_redirect' ) && ! doing_action( 'template_redirect' ) ) {
+		_doing_it_wrong( __FUNCTION__, 'New Relic&#8217;s browser tracking can only be disabled at or before the `template_redirect` action.', '1.0' );
+		return;
+	}
+
+	if ( function_exists( 'newrelic_disable_autorum' ) ) {
+		newrelic_disable_autorum();
+	}
+}
+
+/**
+ * Add the exact URI to NewRelic tracking but only if we're not in the admin
+ */
+function wpcom_vip_add_URI_to_newrelic(){
+	if ( ! is_admin() && function_exists( 'newrelic_add_custom_parameter' ) ){
+		newrelic_add_custom_parameter( 'REQUEST_URI', isset( $_SERVER['REQUEST_URI'] ) ? $_SERVER['REQUEST_URI'] : '' );
+		newrelic_add_custom_parameter( 'HTTP_REFERER', isset( $_SERVER['HTTP_REFERER'] ) ? $_SERVER['HTTP_REFERER'] : '' );
+		newrelic_add_custom_parameter( 'HTTP_USER_AGENT', isset( $_SERVER['HTTP_USER_AGENT'] ) ? $_SERVER['HTTP_USER_AGENT'] : '' );
+	}
+}
+add_action( 'muplugins_loaded', 'wpcom_vip_add_URI_to_newrelic' );
+
+/**
+ * Name cron correctly in New Relic and do not count it as part of the Apdex score.
+ *
+ * We don't want to count cron as part of the apdex because it is not a user facing function and if cron tasks are slow it doesn't imply that the site's performance is impacted. Without removing these, long running cron tasks could flag the site as having performance problems, which would cause false positives in the monitoring.
+ */
+function wpcom_vip_cron_for_newrelic(){
+	if ( defined( 'DOING_CRON' ) && DOING_CRON && function_exists( 'newrelic_ignore_apdex' ) && function_exists( 'newrelic_name_transaction' ) ){
```

Comment:
I missed the addition in 4.8, but we can check `wp_doing_cron()` instead of using the constant. We don't have any sites running < 4.8, so I don't think a `function_exists()` check is needed.